apps/explorer/src/pages/transaction-result/SendReceiveView.tsx

```diff
@@ -1,56 +1,239 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
+
+import { useQuery } from '@tanstack/react-query';
 import cl from 'clsx';
+import { useState, useContext } from 'react';
 
+import { ReactComponent as DoneIcon } from '../../assets/SVGIcons/16px/CheckFill.svg';
+import { ReactComponent as StartIcon } from '../../assets/SVGIcons/Start.svg';
 import Longtext from '../../components/longtext/Longtext';
+import { NetworkContext } from '../../context';
+import { DefaultRpcClient as rpc } from '../../utils/api/DefaultRpcClient';
+import { parseObjectType } from '../../utils/objectUtils';
 
 import styles from './SendReceiveView.module.css';
 
+import { useFormatCoin, CoinFormat } from '~/hooks/useFormatCoin';
+import { Heading } from '~/ui/Heading';
+
 type TxAddress = {
-    timestamp_ms?: number;
     sender: string;
     recipient?: string[];
+    amount?: bigint[];
+    objects?: string[];
 };
-//TODO: Add date format function
-function SendRecieveView({ data }: { data: TxAddress }) {
+
+const getObjType = (objId: string, network: string) =>
+    rpc(network)
+        .getObject(objId)
+        .then((obj) => parseObjectType(obj));
+
+function MultipleRecipients({ sender, recipient, amount, objects }: TxAddress) {
+    const [network] = useContext(NetworkContext);
+
+    const [isSingleCoin, setIsSingleCoin] = useState(false);
+
+    const { data: coinList, isSuccess } = useQuery({
+        queryKey: ['get-coin-types-for-pay-tx', network, objects],
+        queryFn: async () => {
+            const coinList = await Promise.all(
+                objects!.map((objId) => getObjType(objId, network))
+            );
+
+            if (coinList.every((val) => val === coinList[0])) {
+                setIsSingleCoin(true);
```

Comment:
This should be derived out of the response, not set into state in the fetch function.

e.g. `const isSingleCoint = coinList.every((val) => val === coinList[0]);`